# Metagenomics project
## Functional consequences of genomic and single nucleotide variation in infant gut microbiome time series 

### Abstract 

In this study, we explore the intricate dynamics of microbial succession within the infant gut microbiome during the first year
of life, utilizing High-Resolution Metagenomics (HRM) to elucidate the adaptability and persistence of crucial microbial strains.
Specifically, we focus on three pivotal species for infant gut development: Bifidobacterium longum, Phocaeicola (formely Bacteroides)
dorei/vulgatus, and Escherichia coli, asking whether and how these
strains adapt to the conditions of the infant gut, and whether this
adaptation continues throughout the important changes of the
early-life period. We analyze shotgun metagenomics sequencing
data from 306 samples collected from a densely sampled first-year-of-life cohort comprising 12 infants. We first employ the IMP3
pipeline for sample selection based on taxonomic profiling, followed
by assemblies retrieval, functional annotations, and reconstructed
genomes. We validate three different SNP-calling tools under various real-case scenarios and Mutect2, outperforming the others
in sensitivity, is deployed with de novo-reconstructed genomes
to develop an optimized HRM SNP calling workflow subsequently
applied to the data. Our results uncover intriguing dynamics, including a decline in E. coli abundance over time for half of the infants,
increasing and stable trends in P. dorei dorei/vulgatus, and distinct
patterns in B. longum abundance correlating with the introduction of solid food. SNP-level analysis within and between samples
highlights mutations in these species, along with temporal strain
replacement events. Lastly, potential functional consequences of
these changes are assessed and reveal that genes associated with
multi-drug resistance being affected. Additionally, the comparison between reference genome-based (MIDAS2) and metagenome-assembled genome-based (Mutect2) SNP calling methods underscores Mutect2â€™s advantages in detecting within-sample SNPs. By
addressing metagenotyping challenges and opening the path to
Ultra-Resolution Metagenomics (URM), this study contributes to a
deeper understanding of microbial community evolution and development during this critical period, with implications for long-term
health.

### Information
All the scripts that being used for the metagenomics project are provided.

### Chapter 1 - SNPs calling validation (2.4/3.1)*

### Chapter 2 - Per-individual abundances of species of interest over time (2.2/3.2)*

### Chapter 3 - SNP-level analysis (2.5/3.3)*
 
### Chapter 4 - Functional analysis of SNPs (2.6/3.4)*

#### * The corresponding sections in the thesis
A.G.

