# Metagenomics project
## Functional consequences of genomic and single nucleotide variation in infant gut microbiome time series 



### Abstract 

1) This study delves into the dynamic microbial succession within the infant gut microbiome during the first year of life, employing High-Resolution Metagenomics (HRM) to examine the adaptability and persistence of key microbial strains such as Bifidobacterium longum, Phocaeicola (formerly Bacteroides) dorei/vulgatus, and Escherichia coli.

2) Utilizing shotgun metagenomics sequencing data from a densely sampled first-year-of-life cohort comprising 12 infants, our analysis reveals intriguing dynamics, including a decline in E. coli abundance, increasing trends in P. dorei dorei/vulgatus, and distinct patterns in B. longum abundance correlating with the introduction of solid food.

3) Our optimized HRM SNP calling workflow, validated with de novo-reconstructed genomes, highlights temporal strain replacement events and mutations in crucial species, with potential functional consequences such as the impact on genes associated with multi-drug resistance. This study contributes to a deeper understanding of microbial community evolution during this critical period, paving the way for Ultra-Resolution Metagenomics (URM) and offering insights with implications for long-term health.

### Information
All the scripts that have being used for the metagenomics project are provided.

### Chapter 1 - SNPs calling validation (2.4/3.1)*

### Chapter 2 - Per-individual abundances of species of interest over time (2.2/3.2)*

### Chapter 3 - SNP-level analysis (2.5/3.3)*
 
### Chapter 4 - Functional analysis of SNPs (2.6/3.4)*

#### * The corresponding sections in the thesis

<p align="center">
<img src="https://github.com/GArcho44/Thesis_AG/blob/main/fron_page_new.png?raw=true" width=50% height=50%>
</p>

A.G.

