---
title: "SNP.script.1"
author: "Archontis Goumagias"
date: "2022-11-03"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(fig.width=4, fig.height=6, out.width='100%', out.height='100%', fig.align='center')
```

```{r}
# Load appropriate libraries
library(data.table)
library(tidyr)
library(dplyr)
library(ggplot2)
library("Biostrings")
library(ape)
library(readr)
library(stringr)
library(graphics)
library(readxl)
```

```{r}
## Read the data ##

# Read the Mutect2 SNP calling results
mutect2 <- fread(file='mutect2.variants.val.s16.vcf', sep='\t', header = TRUE, skip = '#CHROM')
colnames(mutect2)[1]<- "CONTIG"
mutect2 <- as_tibble(mutect2)

# Read the BCFtools SNP calling results
bcf <- fread(file='bcftools.variants.val.s16.vcf', sep='\t', header = TRUE, skip = '#CHROM')
colnames(bcf)[1]<- "CONTIG"
bcf <- as_tibble(bcf)

# Read the HaplotypeCaller SNP calling results
hc <- fread(file='HCaller.variants.val.s16.vcf', sep='\t', header = TRUE, skip = '#CHROM')
colnames(hc)[1]<- "CONTIG"
hc <- as_tibble(hc)

# Read the recorded SNPs 
positions_1 <- read_csv("positions1a.csv")
positions_2 <- read_csv("positions2a.csv")
positions_3 <- read_csv("positions3a.csv")
pos_all <- full_join(positions_1, positions_2) %>% full_join(positions_3) %>% select(CONTIG, Mutated.sites)

# Read contigs fasta
df <- readDNAStringSet(file="contigs.bin3.fasta")
```

```{r}
# Remove the special characters
pos_all$Mutated.sites <- str_replace_all(pos_all$Mutated.sites, "[[:punct:]]", "")
# Split the character string
pos_all$Mutated.sites <- str_split(pos_all$Mutated.sites, " ")
# Convert the character vectors to numeric ones
pos_all$Mutated.sites <- sapply(pos_all$Mutated.sites, as.numeric)
# Bring the data to the correct form for comparison
pos_all <- pos_all %>% unnest(Mutated.sites) %>% unique() %>% mutate(POS = Mutated.sites) %>% select(CONTIG, POS)
# Correct the positions
pos_all$POS <- pos_all$POS + 1

```

```{r}
## Filtering ##

# Filter for non-SNP variants
mutect2 <- mutect2 %>% filter(REF=='A'|REF=='T'|REF=='G'|REF=='C') %>% filter(ALT=='A'|ALT=='T'|ALT=='G'|ALT=='C')
bcf <- bcf %>% filter(REF=='A'|REF=='T'|REF=='G'|REF=='C') %>% filter(ALT=='A'|ALT=='T'|ALT=='G'|ALT=='C')
hc <- hc %>% filter(REF=='A'|REF=='T'|REF=='G'|REF=='C') %>% filter(ALT=='A'|ALT=='T'|ALT=='G'|ALT=='C')

# Filter columns of interest
mutect2 <- mutect2 %>% select(CONTIG, POS)
bcf <- bcf %>% select(CONTIG, POS)
hc <- hc %>% select(CONTIG, POS)
```

```{r}
## Validation ##

# Calculate the number of reads to simulate
# reads_c <- read_tsv("mg.assembly.contig_coverage.txt", col_names=FALSE) %>% filter(X1 %in% contigs.list) 
# n <- sum (reads_c$X4)
```

```{r}
# Metrics for Mutect2
true_positives_m <- nrow(inner_join(pos_all, mutect2))
true_negatives_m <- sum(df@ranges@width)
false_negatives_m <- nrow(pos_all) - true_positives_m
false_positives_m <- nrow(mutect2) - true_positives_m

# Calculate accuracy, sensitivity, precision
accuracy_m <- (true_positives_m + true_negatives_m) / (true_positives_m + true_negatives_m + false_positives_m + false_negatives_m)
sensitivity_m <- true_positives_m / (true_positives_m + false_negatives_m)
precision_m <- true_positives_m / (true_positives_m + false_positives_m)

```

```{r}
# Metrics for BCFtools
true_positives_b <- nrow(inner_join(pos_all, bcf))
true_negatives_b <- sum(df@ranges@width)
false_negatives_b <- nrow(pos_all) - true_positives_b
false_positives_b <- nrow(bcf) - true_positives_b

# Metrics
accuracy_b <- (true_positives_b + true_negatives_b) / (true_positives_b + true_negatives_b + false_positives_b + false_negatives_b)
sensitivity_b <- true_positives_b / (true_positives_b + false_negatives_b)
precision_b <- true_positives_b / (true_positives_b + false_positives_b)

```

```{r}
# Metrics for HaplotypeCaller
true_positives_h <- nrow(inner_join(pos_all, hc))
true_negatives_h <- sum(df@ranges@width)
false_negatives_h <- nrow(pos_all) - true_positives_h
false_positives_h <- nrow(hc) - true_positives_h

# Metrics
accuracy_h <- (true_positives_h + true_negatives_h) / (true_positives_h + true_negatives_h + false_positives_h + false_negatives_h)
sensitivity_h <- true_positives_h / (true_positives_h + false_negatives_h)
precision_h <- true_positives_h / (true_positives_h + false_positives_h)
```

