---
title: "SNP.script.1"
author: "Archontis Goumagias"
date: "2022-11-03"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(fig.width=12, fig.height=16, out.width='100%', out.height='50%', fig.align='center')
```

```{r}
# Load appropriate libraries
library(data.table)
library(tidyr)
library(dplyr)
library(ggplot2)
library("Biostrings")
library(ape)
library(readr)
library(readxl)
library(RColorBrewer)
library(stringr)
library(viridis)
library(scales)
library(cowplot)
library(ggpubr)
library(grid)
```

```{r}
## Read the data ##

# Read the data with the SNPs number per sample per individual
number_data <- read_xlsx("data/number_of_snps_e.coli.xlsx")
number_data$Individual <- as.factor(number_data$Individual)

# Read the data with the abundance information
ab_data <-  read_xlsx("data/abundance_data.xlsx")
```

```{r}
## ID2 ##

## Make the super-plot that shows allele frequencies, number of SNPs and abundance (per microorganism & individual overtime) ##

# Read the data
merged_data_mutect2 <- read.csv("./data/snps_ID2_mutect2_e.coli.csv")
merged_data_midas <- read.csv("./data/snps_ID2_midas_e.coli.csv")

# Create vectors containing time points information
timepoints <- c(91,126,156,185,306,310,316,317,327,334) # (+)
timepoints_seen <- c(91,126,156,185,306,334) # (+)
timepoints_named <- c("(T1","(T2","(T3","(T4","(T5","(T10") # (+)
timepoints_named.1 <- paste("T",1:10,sep="") # (+)

# Also, get the abundance information and store it into a new dataframe
full_report2 <- ab_data %>% filter(Sample.ID == "ID2") # (+)
ab_info <- full_report2 %>% filter(Bacterial_species=="Escherichia coli") %>% filter(Samples %in% timepoints) %>% filter(Counts >= 50)

# Get the number of reads
reads <- number_data[9:18,4] 
Reads <- reads$Reads

# Convert to numeric
merged_data_midas$SNPs <- as.numeric(merged_data_midas$SNPs)

# Make violin plots for Mutect2
main_plot_mutect2 <- ggplot(data=merged_data_mutect2, mapping=aes(x=timepoint,y=perc)) + 
  geom_violin(aes(fill=SNPs),width=1) +
  theme_classic() + 
  scale_fill_viridis(option="viridis", begin=0, end=0.5, direction=-1) + 
  theme_classic() +
  ggtitle("ID2") + 
  ylab("MAF") + xlab("") +
  theme(plot.title=element_text(hjust = 0.5), legend.position="right") +
  scale_x_discrete(labels=timepoints_named.1)

# Make violin plots for MIDAS
main_plot_midas <- ggplot(data=merged_data_midas, mapping=aes(x=timepoint,y=minor.freq)) + 
  geom_violin(aes(fill=SNPs),width=1)  + 
  scale_fill_viridis(option="inferno", begin=0.2, end=0.8, direction=-1) + 
  ylab("MAF") + xlab("") + ylim(0,1) +
  theme(plot.title=element_text(hjust = 0.5), legend.position="right") +
  scale_x_discrete(labels=timepoints_named.1) +
  theme_classic()

# Create a new plot with the box outside the plot panel
box_plot <- ggplot() + 
  geom_rect(aes(xmin=0,xmax=365,ymin=-0.5,ymax=0),fill="white",col="black") + 
  xlab("Day") + 
  geom_segment(aes(x = timepoints, y = -0.5, xend = timepoints, yend = 0), color = "black") + 
  geom_point(aes(x = timepoints, y = -0.25, size = Reads), color = "#440154", alpha = 0.5)  +
  theme(legend.position="bottom", axis.title.y=element_blank(), axis.text.y=element_blank(), panel.grid = element_blank(), panel.background = element_blank(), axis.ticks = element_blank()) +
  scale_x_continuous(breaks = timepoints_seen, labels=paste(timepoints_named,timepoints_seen,sep=") ")) +guides(fill=guide_legend(title="Number of reads"))

# Make the final plot
p1 <- plot_grid(main_plot_mutect2, main_plot_midas, box_plot, nrow=3, rel_heights = c(1,1, 0.5)) 

```

```{r}
## ID9 ##
## Make the super-plot that shows allele frequencies, number of SNPs and abundance (per microorganism & individual overtime) ##

# Read the data
merged_data_mutect2 <- read.csv("./data/snps_ID9_mutect2_e.coli.csv")

# Create vectors containing time points information
timepoints <- c(10,13,309, 312, 313, 315, 328, 329, 330) # (+)
timepoints_seen <- c(10,309,330) # (+)
timepoints_named <- c("(T1","(T3","(T9") # (+)
timepoints_named.1 <- paste("T",1:9,sep="") # (+)

# Also, get the abundance information and store it into a new dataframe
full_report9 <- ab_data %>% filter(Sample.ID == "ID9") # (+)
ab_info <- full_report9 %>% filter(Bacterial_species=="Escherichia coli") %>% filter(Samples %in% timepoints) %>% filter(Counts >= 50)


# Make violin plots
main_plot <- ggplot(data=merged_data_mutect2, mapping=aes(x=timepoint,y=perc)) + 
  geom_violin(aes(fill=SNPs),width=1) + 
  scale_fill_viridis(option="viridis", begin=0, end=0.5, direction=-1) + 
  theme_classic() +
  ggtitle("ID9") + 
  ylab("") + xlab("") + 
  theme(plot.title=element_text(hjust = 0.5), legend.position="right") +
  scale_x_discrete(labels=timepoints_named.1) 
    
# Create a new plot with the box outside the plot panel
box_plot <- ggplot() + 
  geom_rect(aes(xmin=0,xmax=365,ymin=-0.5,ymax=0),fill="white",col="black") + 
  xlab("Day") + 
  geom_segment(aes(x = timepoints, y = -0.5, xend = timepoints, yend = 0), color = "black") + 
  geom_point(data = ab_info, aes(x = Samples, y = -0.25, size = Counts), color = "#440154", alpha = 0.5) + 
  theme(legend.position="bottom", axis.title.y=element_blank(), axis.text.y=element_blank(), panel.grid = element_blank(), panel.background = element_blank(), axis.ticks = element_blank()) +
  scale_x_continuous(breaks=timepoints_seen, labels=paste(timepoints_named,timepoints_seen,sep=") "))

# Arrange the two plots using the cowplot package
p2 <- plot_grid(main_plot, box_plot, NULL, nrow = 3, rel_heights = c(1, 0.5, 1)) + geom_segment(aes(x = 10, y = -0.5, xend = 10, yend = 0, color="black"))

```

```{r}
## Read the data ##

# Read the data with the SNPs number per sample per individual
number_data <- read_xlsx("data/number_of_snps_b.longum.xlsx")
number_data$Individual <- as.factor(number_data$Individual)

# Read the data with the abundance information
ab_data <-  read_xlsx("data/abundance_data.xlsx")
```

```{r}
## ID2 ##
## Make the super-plot that shows allele frequencies, number of SNPs and abundance (per microorganism & individual overtime) ##

# Read the data
merged_data_mutect2 <- read.csv("./data/snps_ID2_mutect2_b.longum.csv")
merged_data_midas <- read.csv("./data/snps_ID2_midas_b.longum.csv")

# Create vectors containing time points information
timepoints <- c(91,126,185,306,310,311,313,316,317,327,334) # (+)
timepoints_seen <- c(91,126,185,306,334) # (+)
timepoints_named <- c("(T1","(T2","(T3","(T4","(T11")
timepoints_named.1 <- paste("T",1:11,sep="") # (+)

# Also, get the abundance information and store it into a new dataframe
full_report1 <- ab_data %>% filter(Sample.ID == "ID2") # (+)
ab_info <- full_report1 %>% filter(Bacterial_species=="Bifidobacterium longum") %>% filter(Samples %in% timepoints) %>% filter(Counts >= 50)

# some additional processing
ab_info <- ab_info[-5,]
ab_info <- ab_info[-7,]
ab_info$Subsp <- number_data$Subsp[number_data$Individual==2]

# Make violin plots for Mutect2
main_plot_mutect2 <- ggplot(data=merged_data_mutect2, mapping=aes(x=timepoint,y=perc)) + 
  geom_violin(aes(fill=SNPs),width=1) +
  scale_fill_viridis(option="viridis", begin=0.25, end=0.75, direction=-1) + 
  theme_classic() +
  ggtitle("ID2") + 
  ylab("MAF") + xlab("") +
  theme(plot.title=element_text(hjust = 0.5), legend.position="right") +
  scale_x_discrete(labels=timepoints_named.1) 

# Make violin plots for MIDAS
main_plot_midas <- ggplot(data=merged_data_midas, mapping=aes(x=timepoint,y=minor.freq)) + 
  geom_violin(aes(fill=SNPs),width=1)  + ylim(0,1) +
  scale_fill_viridis(option="inferno", begin=0.2, end=0.8, direction=-1) + 
  ylab("MAF") + xlab("") +
  theme(plot.title=element_text(hjust = 0.5), legend.position="right") +
  scale_x_discrete(labels=timepoints_named.1) +
  theme_classic()

# Create a new plot with the box outside the plot panel
box_plot <- ggplot() + 
  geom_rect(aes(xmin=0,xmax=365,ymin=-0.5,ymax=0),fill="white",col="black") + 
  xlab("Day") + 
  geom_segment(aes(x = timepoints, y = -0.5, xend = timepoints, yend = 0), color = "black") + 
  geom_point(data = ab_info, aes(x = Samples, y = -0.25, size = Counts), color = "#21918c", alpha = 0.5) + 
  geom_point(data = number_data[1:11,], aes(x= Day, y = 0, shape=Subsp)) +
  theme(legend.position="bottom", axis.title.y=element_blank(), axis.text.y=element_blank(), panel.grid = element_blank(), panel.background = element_blank(), axis.ticks = element_blank()) +
  scale_x_continuous(breaks = timepoints_seen, labels = paste(timepoints_named, timepoints_seen, sep = ") "))

# Make the final plot
p3 <- plot_grid(main_plot_mutect2, main_plot_midas, box_plot, nrow=3, rel_heights = c(1,1, 0.5)) 

```

```{r}
## ID9 ##
## Make the super-plot that shows allele frequencies, number of SNPs and abundance (per microorganism & individual overtime) ##

# Read the data
merged_data_mutect2 <- read.csv("./data/snps_ID9_mutect2_b.longum.csv")
merged_data_midas <- read.csv("./data/snps_ID9_midas_b.longum.csv")

# Create vectors containing time points information
timepoints <- c(13,65,79,94,111,124,154,184,309,310,311,312,313,315,328) # (+)
timepoints_seen <- c(13,65,124,154,184,309) # (+)
timepoints_named <- c("(T1","(T2","(T6","(T7","(T8","(T9")
timepoints_named.1 <- paste("T",1:15,sep="") 

# Also, get the abundance information and store it into a new dataframe
full_report1 <- ab_data %>% filter(Sample.ID == "ID9") # (+)
ab_info <- full_report1 %>% filter(Bacterial_species=="Bifidobacterium longum") %>% filter(Samples %in% timepoints) %>% filter(Counts >= 50)

# some additional processing
ab_info$subsp <- number_data$Subsp[number_data$Individual==9] # (+)

# Make violin plots for Mutect2
main_plot_mutect2 <- ggplot(data=merged_data_mutect2, mapping=aes(x=timepoint,y=perc)) + 
  geom_violin(aes(fill=SNPs),width=1) +
  theme_classic() +
  scale_fill_viridis(option="viridis", begin=0.25, end=0.75, direction=-1) + 
  ggtitle("ID9") + 
  ylab("MAF") + xlab("") +
  theme(plot.title=element_text(hjust = 0.5), legend.position="right") +
  scale_x_discrete(labels=timepoints_named.1) 

# Make violin plots for MIDAS
main_plot_midas <- ggplot(data=merged_data_midas, mapping=aes(x=timepoint,y=minor.freq)) + 
  geom_violin(aes(fill=SNPs),width=1)  + ylim(0,1) +
  scale_fill_viridis(option="inferno", begin=0.2, end=0.8, direction=-1) + 
  ylab("") + xlab("") +
  theme(plot.title=element_text(hjust = 0.5), legend.position="right") +
  scale_x_discrete(labels=timepoints_named.1) +
  theme_classic()

# Create a new plot with the box outside the plot panel
box_plot <- ggplot() + 
  geom_rect(aes(xmin=0,xmax=365,ymin=-0.5,ymax=0),fill="white",col="black") + 
  xlab("Day") + 
  geom_segment(aes(x = timepoints, y = -0.5, xend = timepoints, yend = 0), color = "black") + 
  geom_point(data = ab_info, aes(x = Samples, y = -0.25, size = Counts), color = "#21918c", alpha = 0.5) + 
  geom_point(data = number_data[42:56,], aes(x= Day, y = 0, shape=Subsp)) +
  theme(legend.position="bottom", axis.title.y=element_blank(), axis.text.y=element_blank(), panel.grid = element_blank(), panel.background = element_blank(), axis.ticks = element_blank()) +
  scale_x_continuous(breaks = timepoints_seen, labels = paste(timepoints_named, timepoints_seen, sep = ") "))

# Make the final plot
p4 <- plot_grid(main_plot_mutect2, main_plot_midas, box_plot, nrow=3, rel_heights = c(1,1, 0.5)) 
```

```{r}
## Read the data ##

# Read the data with the SNPs number per sample per individual
number_data <- read_xlsx("data/number_of_snps_p.dorei.xlsx")
number_data$Individual <- as.factor(number_data$Individual)

# Read the data with the abundance information
ab_data <-  read_xlsx("data/abundance_data.xlsx")
```

```{r}
## ID5 ##
## Make the super-plot that shows allele frequencies, number of SNPs and abundance (per microorganism & individual overtime) ##

# Read the data
merged_data_mutect2 <- read.csv("./data/snps_ID5_mutect2_p.dorei.csv")
merged_data_midas <- read.csv("./data/snps_ID5_midas_p.dorei.csv")

# Create vectors containing time points information
timepoints <- c(11,32,62,92,122,152,212,244,272,298,314,321,321,332,363) # (+)
timepoints_seen <- c(11,32,62,92,122,152,212,244,272,298,363) # (+)
timepoints_named <- c("(T1","(T2","(T3","(T4","(T5","(T6","(T7","(T8","(T9","(T10","(T15")
timepoints_named.1 <- paste("T",1:15,sep="") # (+)

# Also, get the abundance information and store it into a new dataframe
full_report1 <- ab_data %>% filter(Sample.ID == "ID5") # (+)
ab_info <- full_report1 %>% filter(Bacterial_species=="Bacteroides dorei/vulgatus") %>% filter(Samples %in% timepoints) %>% filter(Counts >= 10)

# some additional processing
ab_info$Species <- number_data$Species[number_data$Individual==5]

# Make violin plots for Mutect2
main_plot_mutect2 <- ggplot(data=merged_data_mutect2, mapping=aes(x=timepoint,y=perc)) + 
  geom_violin(aes(fill=SNPs),width=1) +
  scale_fill_viridis(option="viridis", begin=0.5, end=1, direction=-1) + 
  theme_classic() +
  ggtitle("ID5") + 
  ylab("MAF") + xlab("") +
  theme(plot.title=element_text(hjust = 0.5), legend.position="right") +
  scale_x_discrete(labels=timepoints_named.1) +
    scale_color_continuous(limits = c(0, 1000000))

# Make violin plots for MIDAS
main_plot_midas <- ggplot(data=merged_data_midas, mapping=aes(x=timepoint,y=minor.freq)) + 
  geom_violin(aes(fill=SNPs),width=1)  + ylim(0,1) +
  scale_fill_viridis(option="inferno", begin=0.2, end=0.8, direction=-1) + 
  ylab("MAF") + xlab("") +
  theme(plot.title=element_text(hjust = 0.5), legend.position="right") +
  scale_x_discrete(labels=timepoints_named.1) +
  theme_classic()

# Create a new plot with the box outside the plot panel
box_plot <- ggplot() + 
  geom_rect(aes(xmin=0,xmax=365,ymin=-0.5,ymax=0),fill="white",col="black") + 
  xlab("Day") + 
  geom_segment(aes(x = timepoints, y = -0.5, xend = timepoints, yend = 0), color = "black") + 
  geom_point(data = ab_info, aes(x = Samples, y = -0.25, size = Counts), color = "#fde725", alpha = 0.5) + 
  geom_point(data = number_data[26:40,], aes(x= Day, y = 0, shape=Species)) +
  theme(legend.position="bottom", axis.title.y=element_blank(), axis.text.y=element_blank(), panel.grid = element_blank(), panel.background = element_blank(), axis.ticks = element_blank()) +
  scale_x_continuous(breaks = timepoints_seen, labels = paste(timepoints_named, timepoints_seen, sep = ") "))

# Make the final plot
p5 <- plot_grid(main_plot_mutect2, main_plot_midas, box_plot, nrow=3, rel_heights = c(1,1, 0.5)) 
```

```{r}
## ID9 ##
## Make the super-plot that shows allele frequencies, number of SNPs and abundance (per microorganism & individual overtime) ##

# Read the data
merged_data <- read.csv("./data/snps_ID9_mutect2_p.dorei.csv")

# Create vectors containing time points information
timepoints <- c(13,184,244,274,309,310,311,313,315,329,330,334) # (+)
timepoints_seen <- c(13,184,244,274,309,334) # (+)
timepoints_named <- c("(T1","(T2","(T3","(Τ4","(T5","(T12")
timepoints_named.1 <- paste("T",1:12,sep="") # (+)

# Also, get the abundance information and store it into a new dataframe
full_report1 <- ab_data %>% filter(Sample.ID == "ID9") # (+)
ab_info <- full_report1 %>% filter(Bacterial_species=="Bacteroides dorei/vulgatus") %>% filter(Samples %in% timepoints) %>% filter(Counts >= 10)

# some additional processing
ab_info$Species <- number_data$Species[number_data$Individual==9]

# Make violin plots
main_plot <- ggplot(data=merged_data, mapping=aes(x=timepoint,y=perc)) + 
  geom_violin(aes(fill=SNPs),width=1) + 
  scale_fill_viridis(option="viridis", begin=0.5, end=1, direction=-1) + 
  theme_classic() +
  ggtitle("ID9") + 
  ylab("") + xlab("") +
  theme(plot.title=element_text(hjust = 0.5), legend.position="right") +
  scale_x_discrete(labels=timepoints_named.1) 

# Create a new plot with the box outside the plot panel
box_plot <- ggplot() + 
  geom_rect(aes(xmin=0,xmax=365,ymin=-0.5,ymax=0),fill="white",col="black") + 
  xlab("Day") + 
  geom_segment(aes(x = timepoints, y = -0.5, xend = timepoints, yend = 0), color = "black") + 
  geom_point(data = ab_info, aes(x = Samples, y = -0.25, size = Counts), color = "#fde725", alpha = 0.5) + 
  geom_point(data = number_data[53:64,], aes(x= Day, y = 0, shape=Species)) +
  theme(legend.position="bottom", axis.title.y=element_blank(), axis.text.y=element_blank(), panel.grid = element_blank(), panel.background = element_blank(), axis.ticks = element_blank()) +
  scale_x_continuous(breaks = timepoints_seen, labels = paste(timepoints_named, timepoints_seen, sep = ") "))

# Arrange the two plots using the cowplot package
p6 <- plot_grid(main_plot,box_plot, NULL, nrow = 3, rel_heights = c(1, 0.5, 1)) + geom_segment(aes(x = 10, y = -0.5, xend = 10, yend = 0, color="black"))

```

```{r}
# Create & save the final plot as tiff image

tiff("SNPs_viz_results.tiff", units="in", width=20, height=20, res=300)

plot_grid(p1,p2,p3,p4,p5,p6, nrow=3, labels = c("A","","B","","C",""))

dev.off()
```

